File: e2e-apps/ruby-rails-web/app/controllers/dashboard_controller.rb
# frozen_string_literal: true

require "erb"

class DashboardController
  def show(request:, session_user:)
    if session_user.empty?
      return {
        status: 302,
        location: "/login",
        body: ""
      }
    end

    escaped_user = ERB::Util.html_escape(session_user)

    {
      status: 200,
      content_type: "text/html",
      body: <<~HTML
        <section style="min-height: 100vh; padding: 2.5rem 1.25rem; background: #fdf2f8; font-family: 'Helvetica Neue', Arial, sans-serif;">
          <div style="max-width: 720px; margin: 0 auto; background: #ffffff; border-radius: 20px; padding: 2.25rem; box-shadow: 0 25px 70px rgba(99, 102, 241, 0.25);">
            <div style="font-size: 11px; font-weight: 700; letter-spacing: 0.2em; text-transform: uppercase; color: #7c3aed;">Ruby Browser Signal R1</div>
            <h1 style="margin: 0.75rem 0 0.25rem; font-size: 2.25rem; color: #4c1d95;">Dashboard</h1>
            <p id="session-user" style="margin: 0 0 1rem; font-size: 1rem; font-weight: 600; color: #312e81;">Signed in as #{escaped_user}</p>
            <p style="margin: 0 0 1rem; font-size: 1.05rem; color: #1f2937;">Seeded alerts are ready.</p>
            <p style="margin: 0 0 0.75rem; font-size: 0.95rem; line-height: 1.6; color: #374151;">
              Autonomous code agents read the issue instructions, make the smallest safe edit, and run the requested validation command before reporting the outcome. The dashboard surface documents that workflow so the next reviewer easily understands what changed and why.
            </p>
            <p style="margin: 0; font-size: 0.95rem; line-height: 1.6; color: #4b5563;">
              A healthy run keeps the edit scope tight, executes the requested checks, and publishes artifacts so the repair remains auditable and reproducible.
            </p>
          </div>
        </section>
      HTML
    }
  end
end

File: e2e-apps/java-spring-web/src/main/java/example/web/DashboardController.java
package example.web;

public class DashboardController {
    public ResponsePlan dashboard(String userEmail) {
        if (userEmail == null || userEmail.isBlank()) {
            return ResponsePlan.redirect("/login");
        }
        String escapedUser = escapeHtml(userEmail);
        return ResponsePlan.html(
            200,
            """
            <section style="min-height: 100vh; padding: 48px; background: #f4e7d3; color: #3f2f21; font-family: Georgia, 'Times New Roman', serif;">
                <div style="max-width: 760px; margin: 0 auto; background: rgba(255, 249, 240, 0.92); border: 1px solid #dcc7aa; border-radius: 18px; padding: 32px; box-shadow: 0 18px 40px rgba(95, 70, 40, 0.12);">
                    <div style="font-size: 14px; font-weight: 700; letter-spacing: 0.08em; text-transform: uppercase; color: #8a5a2b;">Java Browser Signal J1</div>
                    <h1 style="margin: 16px 0 12px; font-size: 40px;">Dashboard</h1>
                    <div id="session-user" style="margin-bottom: 20px; font-size: 16px;">Signed in as %s</div>
                    <div style="margin-bottom: 20px; font-size: 15px; font-weight: 600; color: #6d4c2c;">Seeded alerts are ready.</div>
                    <p style="margin: 0 0 16px; font-size: 18px; line-height: 1.7;">Autonomous code agents read issue instructions, edit only the allowed files, run checks, and report their progress back on GitHub. They keep the work scoped so each change lines up with the task contract and leaves a clear audit trail.</p>
                    <p style="margin: 0 0 16px; font-size: 18px; line-height: 1.7;">Autonomous code agents read the issue instructions, make the smallest safe edit, and run the requested validation command before reporting the outcome. The dashboard surface documents that workflow so the next reviewer easily understands what changed and why.</p>
                    <p style="margin: 0 0 16px; font-size: 18px; line-height: 1.7;">A healthy run keeps the edit scope tight, executes the requested checks, and publishes artifacts so the repair remains auditable and reproducible.</p>
                    <p style="margin: 0; font-size: 18px; line-height: 1.7;">That workflow helps teams clear routine engineering work faster because small fixes move forward with less manual coordination, while the guardrails keep each automated pass focused, reviewable, and easy to trust.</p>
                </div>
            </section>
            """.formatted(escapedUser)
        );
    }

    private static String escapeHtml(String raw) {
        return String.valueOf(raw)
            .replace("&", "&amp;")
            .replace("<", "&lt;")
            .replace(">", "&gt;")
            .replace("\"", "&quot;")
            .replace("'", "&#x27;");
    }
}
